### ISSUE TITLE
rendering highway=escape

### ISSUE THREAD

**Issue by al--** [gravitystorm/openstreetmap-carto#1239]
Please render highway=escape. Style could be much like highway=service but with a dashed border.

[highway=escape tagged in Austria](http://overpass-turbo.eu/s/76p)
Wiki: tag [highway=escape](http://wiki.openstreetmap.org/wiki/Tag:highway%3Descape)

tnx /al

**Comment by mboeringa** [comment]
> Please render highway=escape.

What is the problem? The feature is rendered and visible on the map... or are you just asking for a special styling, in which case you might consider rephrasing the title and contents of this issue. 

Something like "Apply specialized styling for highway=escape" maybe?

**Comment by HolgerJeromin** [comment]
http://www.openstreetmap.org/way/259526665 is not rendered. And I do not find the word escape in this repository.

**Comment by mboeringa** [comment]
> http://www.openstreetmap.org/way/259526665 is not rendered. And I do not find the word escape in this repository.

Am I just going nuts here? Because I saw lines yesterday - even drawn with labels - when I checked @al--'s Overpass Turbo link, and clicked from there to the main OpenStreetMap website...

**Comment by mkoniecz** [comment]
@mboeringa No, that is just delay in rendering. http://www.openstreetmap.org/way/301772151/history used to be tagged as highway=service - so on old tiles it was displayed.

Flushing cache in browser resulted in disappearing road.

**Comment by mkoniecz** [comment]
Maybe rendering it exactly like [highway=service; service=parking_aisle] would be a good idea?

**Comment by BorutAtGit** [comment]
Something more distinctive would perhaps also be a good idea. See for example what traffic sign is being used for this in Switzerland: http://de.wikipedia.org/wiki/Notfallspur_(Gefälle)

Special apps will most probably become available for professional truckers anyway, but a prominent rendering here would help spread the word among mappers. (As this is about a life saving utility, virtually any rendering will be better than none.)

**Comment by mkoniecz** [comment]
BTW, note that less than this tag is used less than 400 times - http://taginfo.openstreetmap.org/tags/highway=escape

**Comment by jgpacker** [comment]
It seems some mappers use `highway=service` + `service=escape_lane` (80 uses right now), but this could be considered "tagging for the renderer" since it doesn't fit with `highway=service`'s definition.

The best would be that this tag is rendered similar to [one of it's signs](https://commons.wikimedia.org/wiki/Category:Emergency_stopping_lane_signs).
I believe rendering it as a double black-and-white dotted line would be a good fit (similar to [this](http://jsfiddle.net/wjoz9gfn/)).

Other than that, the same rendering as a parking aisle would also be ok.

**Comment by althio** [comment]
If we use an existing rendering, I think it should bear the idea of restrictive access (something like access=no/private), not the regular parking aisle. It is not a parking or a rest area.

Better if we use a new rendering. Some kind of checker or similar pattern would be distinctive ([see link of signs](https://commons.wikimedia.org/wiki/Category:Emergency_stopping_lane_signs) from @jgpacker) and useful for this kind of emergency features.

Note that this feature is international, useful, with signs, easy to identify.
Note that the related wiki page is well-documented, useful, easy to tag.
Note that usage will increase with rendering, in particular with distinctive rendering.

**Comment by Rovastar** [comment]
I would be against different rendering only because it is a very niche feature and we cannot render everything. Rendering this could make it more difficult to render more needed features.

**Comment by althio** [comment]
> Rendering this could make it more difficult to render more needed features.

Maybe I misunderstand your point? This escape feature is unlikely to spatially clash with others; its distinctive rendering is unlikely to be used for other features.
OTOH a distinctive rendering is desirable as it is a niche feature and an emergency feature.
Could you please expand on the limitations?

**Comment by gravitystorm** [comment]
> an emergency feature

Nobody, and I do mean nobody, needs to check for escape roads on openstreetmap-carto maps in an emergency situation.

If anyone else suggests that the implementation of this feature request is somehow life-saving then I will lock the issue.

**Comment by gravitystorm** [comment]
> a distinctive rendering is desirable as it is a niche feature

Since it is niche, then what is needed is an unobtrusive rendering. It should not be eye-catching and it should not draw any attention to the feature since very few people will be looking at a openstreetmap-carto map and trying to find where the escape roads are. The only reason to render them is to "fill the gap" where they exist, and to satisfy curiosity.

**Comment by Rovastar** [comment]
Maybe I misunderstand your point? This escape feature is unlikely to spatially clash with others; its distinctive rendering is unlikely to be used for other features.
  OTOH a distinctive rendering is desirable as it is a niche feature and an emergency feature.
  Could you please expand on the limitations?

Like I said we cannot render everything. Rendering with a dashed border as suggested if wrong as there could be other features that are more needed that could use this (toll roads, unpaved roads, etc)
I see no problem with it as a normal service road rendering here. 

  Nobody, and I do mean nobody, needs to check for escape roads on openstreetmap-carto maps in an emergency situation

I thought the same thing and would have replied the same. :)

"Oh, no my breaks have failed. I am traveling at full speed. What do I do?
I know I'll get my phone out and connect to openstreetmap.com and lookup the key for escape roads and find it on the map."

**Comment by althio** [comment]
@Rovastar 
> I would be against different rendering only because it is a very niche feature and we cannot render everything. Rendering this could make it more difficult to render more needed features.
> ...
> Like I said we cannot render everything. Rendering with a dashed border as suggested if wrong as there could be other features that are more needed that could use this (toll roads, unpaved roads, etc)
> ...
> I see no problem with it as a normal service road rendering here.

I cannot yet grasp the rationale so please bear with me.


When is it OK to render it with an existing rendering? Only if the features are similar, otherwise it just make it more difficult to understand the legend and features? But when features are different enough for tagging, we can also consider whether they are different enough for rendering.

Why rendering as a normal service road? This is not a normal service road by definition and usage.

Why is it *not* OK to render it with an existing (or potential) rendering (dashed border?) [because it could be used for toll roads or unpaved roads]. At least that could make some sense since this is (kind of) an unpaved road.

Why is it *not* OK to render it with a new rendering that is unlikely to be used for another feature (like @jgpacker and I are trying to propose)? What is wrong with proposing something different to start with? (And I do NOT mean eye-catching.) The only downside I see is that the legend is more crowded if all specific features are displayed there.


Bottom line: I do not take decision here, I only give my opinion, advice and offer proposals when I can.


For clarity, all considered options with my order of preference:
1. new rendering to be established, unobtrusive and distinctive and not eye-catching and easily associated with feature. [if this is not possible, skip; but at least consider the option].
2. existing rendering, same as `highway=service + access=no` (inside red dashed)
3. 3a. new/existing rendering, variation on service road with dashed border
3. 3b. new/existing rendering, same as unpaved road (about `surface=*` not `highway=*` but good fit)
4. no rendering
5. rendering like normal service road or parking aisle.
6. any obtrusive rendering

I have not seen other proposals yet. I will make a mockup for direct illustration of possible rendering.

**Comment by jgpacker** [comment]
@althio The problem with giving it a new rendering is mostly because it creates more work for the map style's mantainers. Making it the same as a parking aisle would both make it render and make their job easier. Keep in mind that when they make changes in the map, they have to test it, check for conflicts, give a higher priority to more important features, try to make the map look good, etc.
And correct me if I'm wrong, but as far as I know they all volunteer their free time to work here.

Even though a unique rendering would be the best, personally I'm ok with `highway=escape` being rendered as a parking aisle or similar. 
I don't think it need to look like it has restricted access, unless more people want it.

**Comment by althio** [comment]
>> a distinctive rendering is desirable as it is a niche feature

> Since it is niche, then what is needed is an unobtrusive rendering. It should not be eye-catching

This feels like déjà-vu between us ;)
I am sorry if I do not make myself clear but I honestly want to use 'distinctive' with a different meaning than 'obtrusive'.
* distinctive ~ distinguishable ~ different enough
* obtrusive ~ eye-catching ~ prominent

Rendering (eg. patterns, icons...) can be IMO quite distinctive and unobtrusive at the same time. It is a matter of balance between size, shape, color, contrast, opacity, zoom... I would imagine 
* distinctive pattern like checker (see signs) or full dashes (see @jgpacker proposal) 
* and unobtrusive with reduced opacity (0.3? for niche and not regularly accessible) and late enough for zoom (z>18).

> The only reason to render them is to "fill the gap" where they exist, and to satisfy curiosity.

Only reason? Other reasons might be to showcase the richness of the underlying database, to hint at multiple uses or output from the data and of course the usual loop to promote tagging of the rendered feature. Not limited to my list, maybe other people could find other reasons.


[Mockup proposal](http://jsfiddle.net/wjoz9gfn/2/) ![image](https://cloud.githubusercontent.com/assets/9863567/5839720/530d73f0-a190-11e4-9e18-5d5ed6ad86b7.png) with opacity to be adjusted.
Approximate rendering for high zoom only
![escape](https://cloud.githubusercontent.com/assets/9863567/5840554/a256fa20-a196-11e4-8c79-0be9e9332282.png)

**Comment by javbw** [comment]
>Nobody, and I do mean nobody, needs to check for escape roads on openstreetmap-carto maps in an emergency situation. 

It would sure be nice to look at a route and easily note where they are !

These are rare, but lifesaving safety features. Rendering them as a service road is wrong for 3 reasons - 

- They are not something for access. At best they are a kind of barrier, albeit a long one. 
- They are usually something that going into would bad Hitting a gravel pile, getting your car stuck in loose sand is not good) - if they are rendered like a road, then access to something at the end would be assumed, when there is actually nothing to access.
- If we then say "well, put an access restriction on it", then we are communicating to people that access to this is restricted - AKA it is closed to public use. I don't think we'd want to communicate that a feature has restricted access when it in fact is available to people planning to use the route. 

That grey checkerboard render is awesome.  Lets' do it! ^_^

**Comment by mkoniecz** [comment]
> There are power lines rendered on OSM. That is less useful to people

Power lines are excellent orientation points.

> grey checkerboard

Completely new and hard to understand symbol is not going to be used for something used less than 250 times.

**Comment by javbw** [comment]
> Power lines are excellent orientation points.
> 
Determining which point you intersect a line presented to you that is kilometers long can be difficult, without another point of reference. At that point, the tower you are closer to is more of a landmark than the wires itself. It is the wire I object to more than anything.

In a country that uses solid black lines to (exclusively) represent private train lines (the dashed is for the state system only) and is absolutely coated with high voltage power lines - I would venture the confusion between rail lines due to their prominence at z16 and below (to about z16 or 17) heavily outweighs their landmark values - and I live next to a spur of the highest capacity transmission line in the world https://goo.gl/maps/Wqujt <https://goo.gl/maps/Wqujt> 

https://goo.gl/maps/qAnBR <https://goo.gl/maps/kPsvF>  (I’m not basing this on google’s stylesheet for the map - they are following Japan’s historical mapping tendencies)

http://www.openstreetmap.org/#map=14/36.3835/139.1072 <http://www.openstreetmap.org/#map=14/36.3835/139.1072>  (zoomed in a bit)

Wow - the private train lines now run north-south! how useful! 

And for something as small as a runaway truck ramp (escape), having a unique render is useful because it is a *signed* road/motorway feature.  And if it only gets 250 renders, then however we choose to render it (as long as it doesn’t look like a service road) won’t bother as many people as these power lines currently do.  Towers are landmarks - at z16.  The lines are only a landmark at z16 and below. The towers don’t need to be rendered at z14, and the lines are a massive detriment to the map at z14, as they override the soft pastel nature of the roads.  They imply something a lot more solid, useful, and important than a string of wire that does nothing for you getting from A to B (nor impedes you, like a river). seeing a massive road network, train lines, rivers, and the rest of the map at z16 is more helpful for orienteering - as determining what unlabeled, un-refed tower on a map you happen to be near when presented with a string of them.
We should tag them, we should render them, but we need to choose a better cutoff for the render, because at z14 they are unnecessary, confusing clutter. 


People looking down the motorway see the escape and it’s associated sign, but the little black flicker of the lines passing over the motorway  (with no sign to let people make note of it) is a much less useful orienteering marker to people who are diving on roads that would make use of escapes.  if you were somewhere where they have escapes, often it is a hilly area, so you can’t see a string of high voltage lines unless they are on the side of the mountain you can see for that moment - which, again, is much less useful than a known, easily seen, signed, well recognized safety feature built into the *road system itself*! 

There may be only 250 marked, but the 5 I have ever seen are easily remembered, and a landmark while driving.

The escapes are a better landmark for motorway drivers than a string of power lines, or even a single prominent tower.

Javbw=

**Comment by Rovastar** [comment]
*sigh* 
if you have problems with power lines then please make a new issue. There you can discuss the merits of what zoom level and usefulness of it.

Not really sure why this hasn't been closed yet. Are we seriously going to render this?

**Comment by javbw** [comment]
Just responding back to someone who didn't like my example, and I try to backup what I say. 
 
A nice rendering of the escape would be great. 

Javbw

**Comment by jgpacker** [comment]
Was this closed intentionally? 
I respect the decision if it was. I just want to be sure it wasn't accidental.

**Comment by althio** [comment]
I feel sorry for @al-- that this request for rendering is being closed.

There is support for rendering it (I would say maybe 5 or 6 supports out of 7 or 8 opinions, but this is only a discussion, not a voting).
The discussion was furthermore animated about *how* it could be rendered with proposals or positive opinions from 6 contributors at least @al--  @mkoniecz @jgpacker @BorutAtGit @althio @javbw. Maybe it was a bit too philosophical or digressing at times but we were still discussing on how to do it properly and trying to reach a agreement.

On a personal note I would have liked more feedback about the possibilities of new rendering https://github.com/gravitystorm/openstreetmap-carto/issues/1239#issuecomment-70871608, or the particular proposal https://github.com/gravitystorm/openstreetmap-carto/issues/1239#issuecomment-70891908.
One feedback was very positive and the other was "Completely new and hard to understand symbol". I feel we could have a further discussion with a practical case and proposals. With no rush to answer, to accept, to decide or to close.

**Comment by Klumbumbus** [comment]
I just want to add my opinion: please render highway=escape. This is an interesting and important part of the road network.

**Comment by mboeringa** [comment]
I would agree that at least a basic rendering, starting with something similar or equal to highway=services, is a no-brainer. Any sophisticated specialized rendering could be further discussed and developed later.

**Comment by HolgerJeromin** [comment]
a service like rendering would be nice, yes. Much better than not rendering.

**Comment by Rovastar** [comment]
Although I called for closing this, that was mainly for radically different styles am all for rendering as service.

**Comment by polarbearing** [comment]
> Maybe rendering it exactly like [highway=service; service=parking_aisle] would be a good idea?
+1

**Comment by mkoniecz** [comment]
I wrongly thought that consensus is "Either nothing or checkerboard". Still, note that this tag is used really rarely (I think that more energy went into this discussion than into tagging all currently marked highway=escape).

**Comment by mboeringa** [comment]
> I wrongly thought that consensus is "Either nothing or checkerboard". Still, note that this tag is used really rarely (I think that more energy went into this discussion than into tagging all currently marked **highway=escape**).

@mkoniecz, while the tag itself is used rather rarely at the moment, this says little about the actual occurrence of the feature in the database in terms of an existing line element. As per the example in the post of the OP of this thread, some features are already in the database, but tagged otherwise to show up (e.g. highway=service, unclassified etc.), essentially "tagging-for-the-renderer"...

Not rendering **highway=escape**, would thus also imply losing objects that were rendered before when people start using the "real" / right tag, and convert existing tags to the more appropriate one. I think we should actually encourage good tagging, instead of bad, and in this case, a separate specific tag for **highway=escape**, as opposed to general **highway=service**, seems justified. This feature is to distinct to classify it as general **highway=service**.

**Comment by kocio-pl** [comment]
Hm, what about highway=service+service=escape? We already have service=emergency_access after all. Maybe we want the service tag to be reserved only for accessing something, but if this is only matter of too tight definition, this seems to me like a subcategory of a service roads.

**Comment by mboeringa** [comment]
> **Hm, what about highway=service+service=escape?** We already have service=emergency_access after all. Maybe we want the service tag to be reserved only for accessing something, but if this is only matter of too tight definition, this seems to me like a subcategory of a service roads.

That would introduce yet another undocumented tagging scheme... your proposition would require re-tagging and a Wiki change before making sense. I don't think this is sensible now.

**highway=escape** is already documented after all:
http://wiki.openstreetmap.org/wiki/Key:highway (under "Other highway features")
and
http://wiki.openstreetmap.org/wiki/Tag:highway%3Descape

In addition, contrary to all other "service" roads, an escape, as @althio and @javbw also argued, is not for "driving into" under anything but extraordinary conditions. This distinguishes them from **highway=service**, so I think the current practice of a separate tag is kind of OK, even disregarding the documented, and hence, for what it is worth in this introductory stage of a rather new tag, established practice.

**Comment by matkoniecz** [comment]
http://taginfo.openstreetmap.org/tags/highway=escape - used only 351 times. Too low to justify time needed to make changes to rendering style and justify increased complexity of code and map.

**Comment by dieterdreist** [comment]
2015-07-03 17:47 GMT+02:00 Mateusz Konieczny <notifications@github.com>:

> http://taginfo.openstreetmap.org/tags/highway=escape - used only 351
> times. Too low to justify time needed to make changes to rendering style
> and justify increased complexity of code and map.



I am not advocating this tag, but wanted to point to Andy Allan's 2
criteria for new features which do not include usage numbers:
https://github.com/gravitystorm/openstreetmap-carto/issues/1630

**Comment by kocio-pl** [comment]
> The only reason to render them is to "fill the gap" where they exist, 

I would like to see them on some high enough zoom level exactly because having gaps indicates we fail to visually "explain" the structure of given area.

On low and medium scale the need to exclude detailed features from the map is essential, but on high scale it starts being just an old habit and exaggeration. Let's look [on the mock up rendering] (https://github.com/gravitystorm/openstreetmap-carto/issues/1239#issuecomment-70891908) once again:

1. Is the area cluttered in any way? I guess no, because on micromapping level the features are no longer "stacked" and there is typically a lot of free space (totally uncovered gaps or just some general background areas like landuse - both with high entropy).

2. Is the area more complete and logical? Sure - and I think this is a very good thing. Now it's clear to me why the small part of forest has such a strange shape. 

3. Compare the escape road with the grass: which one is more important here? Basically the landuse=grass (unlike the meadow) is a "landcover" tag indeed and is used mainly to show what is visible, no matter what's the function of it (could you enumerate any?). On high scale the escape road is also visible, especially to the drivers - as they see the exit - but are way more important facility, even if specialized, so intentionally omitting it is strange.

> and to satisfy curiosity.

It sounds to me like you see this motivation as inferior - so what are valid reasons to look at the general map? I could not imagine such thing at all and find this kind of attitude to be missing the point. If the map can be complete and clean at the same time, it's clearly better for any reasons (including curiosity).

**Comment by kocio-pl** [comment]
[This is interesting example](https://github.com/gravitystorm/openstreetmap-carto/pull/1635#issue-93131881) of how much different standards we use judging medium and high scale. Are there too much elements or are those pictures just complete (gapless)? If it's bad or good than?

**Comment by printmaps** [comment]
I recommend to draw highway=escape, which is an important emergency infrastructure, as gray line.

https://www.openstreetmap.org/#map=17/50.67673/8.29081

Drawing example (not from osm-carto):

![bildschirmfoto 2017-10-16 um 12 55 39](https://user-images.githubusercontent.com/26689799/31608684-7035c64a-b271-11e7-9050-32872fe825b4.png)

**Comment by matkoniecz** [comment]
It looks like a current airport rendering (compare with http://www.openstreetmap.org/#map=17/50.07857/19.79808 )

**Comment by kocio-pl** [comment]
I don't think this could be a problem, since it's well defined feature and is related to general roads, not the airports.

**Comment by jeisenbe** [comment]
Used 875 times now. It's not a high priority, but if someone is interested in submitting a PR for this issue, it could be reasonable to rendering `highway=escape` the same as minor service roads.

**Comment by cicku** [comment]
Waiting for this feature too.
